indico/modules/events/surveys/controllers/display.py

```diff
@@ -88,8 +93,10 @@ def _process(self):
         form = self._make_form()
         if form.validate_on_submit():
             submission = self._save_answers(form)
-            if submission.is_anonymous:
+            if submission.is_anonymous and submission.user:
                 submission.user = None
+                anonymous_submission = SurveyAnonymousSubmission(survey_id=self.survey.id, user_id=session.user.id)
+                db.session.add(anonymous_submission)
```

Comment:
always use the relationships to link models - and you don't even need the add call that way :)